crates/iota-core/src/authority.rs

```diff
@@ -5287,33 +5287,41 @@ impl AuthorityState {
             );
         }
 
-        let authenticator_id = self.get_dynamic_field_object_id(
+        let authenticator_info_field_id = dynamic_field::derive_dynamic_field_id(
             auth_account_object_id,
-            AuthenticatorInfoV1::tag().into(),
-            AUTHENTICATOR_DF_NAME.as_bytes(),
-        )?;
+            &account::authenticator_df_name_type_tag(),
+            &account::authenticator_df_name_as_bcs_bytes(),
+        )
+        .map_err(|_| UserInputError::UnableToGetMoveAuthenticatorId {
+            account_object_id: auth_account_object_id,
+        })?;
 
-        if let Some(authenticator_id) = authenticator_id {
-            let authenticator_info = self
-                .get_object_cache_reader()
-                .try_find_object_lt_or_eq_version(
-                    authenticator_id,
-                    auth_account_object_seq_number,
-                )?;
+        let authenticator_info_field = self
+            .get_object_cache_reader()
+            .try_find_object_lt_or_eq_version(
+                authenticator_info_field_id,
+                auth_account_object_seq_number,
+            )?;
 
-            if let Some(authenticator_info) = authenticator_info {
-                AuthenticatorInfoV1::try_from(authenticator_info)
-            } else {
-                Err(UserInputError::MoveAuthenticatorNotFound {
-                    authenticator_object_id: authenticator_id,
-                    account_object_id: auth_account_object_id,
-                    account_object_version: auth_account_object_seq_number,
-                }
-                .into())
-            }
+        if let Some(authenticator_info_field_obj) = authenticator_info_field {
+            let field_move_object = authenticator_info_field_obj
+                .data
+                .try_as_move()
+                .expect("dynamic field should never have be a package object");
```

Comment:
Something went wrong here, should it be like this?
```suggestion
                .expect("dynamic field should never be a package object");
```

Reply:
fixed in beaa7a6b00